### PR TITLE
More documentation improvements

### PR DESCRIPTION
**Pull request by chshersh**
Let me know, what do you think, and I will update the `.cabal` description accordingly.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,36 +55,56 @@ the library. So below you can find the key design principles behind `relude`:
...
+
 9. **Exploration.** We have space to experiment with new ideas and proposals
    without introducing breaking changes. `relude` uses the approach with
-   `Extra.*` modules which are not exported by default, so it makes it quite
+   `Extra.*` modules which are not exported by default. The chosen approach makes it quite
    easy for us to provide new functionality without breaking anything and let
    the users decide to use it or not.
 
-<!-- TODO: anti-goals? -->
+In addition to our key design principle, the following list of
```
Comment (vrom911) [inline README.md:97]:
```suggestion
In addition to our key design principles, the following list of
```

File: README.md [hunk 2]
```diff
@@ -326,7 +360,7 @@ Main differences from `Prelude` can be grouped into the following categories:
 * [`unordered-containers`](http://hackage.haskell.org/package/unordered-containers)
 
 If you want to clean up your imports after switching to `relude`, you can use
-`relude`-specific [`.hlint.yaml`](.hlint.yaml) configuration for this task.
+the `relude`-specific [`.hlint.yaml`](.hlint.yaml) configuration for this task.
```
Comment (vrom911) [inline README.md:363]: Can you please expand the link in here as well? :pray:

File: README.md [hunk 3]
```diff
@@ -346,7 +380,7 @@ Multiple sorting functions are available for different use-cases:
 `readMaybe` and `readEither` are similar to `read` but unlike it, they are total
 and return either `Maybe` or `Either` with a parse error.
 
-`(&)` – reverse application. The following three expressions are
+`(&)` is reverse application. The following three expressions are
```
Comment (vrom911) [inline README.md:383]: is "the"/"a"? I am not sure though..

File: README.md [hunk 4]
```diff
@@ -23,13 +23,26 @@ defines the library decisions and might tell you more about the priorities of
...
+   `reludes` strives to encourage best-practices and reduce the
+   chances of introducing a bug.
+
+  | __Partial__                     | __Total__                                  |
+  |---------------------------------|--------------------------------------------|
+  | `head :: [a] -> a`              | `head :: NonEmpty a -> a`                  |
+  | `tail :: [a] -> [a]`            | `tail :: NonEmpty a -> [a]`                |
+  | `read :: Read a => String -> a` | `readMaybe :: Read a => String -> Maybe a` |
+  | `fromJust :: Maybe a -> a`      | `fromMaybe :: a -> Maybe a -> a`           |
+
```
Comment (vrom911) [inline README.md:43]: Great table :heart: 

Do you think it is a good idea to add the same table to the Haddocks of the `Relude.Unsafe` module?
With the disclaimer, that maybe you can avoid unsafe function blah blah?

Comment (chshersh) [reply README.md:43]: @vrom911 I think this is an excellent idea :+1:

File: src/Relude/Unsafe.hs [hunk 5]
```diff
@@ -9,9 +9,25 @@ Maintainer:  Kowainik <xrom.xkov@gmail.com>
...
+| @'head' :: [a] -> a@              | @'Relude.head' :: NonEmpty a -> a@                                  |
++-----------------------------------+---------------------------------------------------------------------+
+| @'tail' :: [a] -> [a]@            | @'Relude.tail' :: NonEmpty a -> [a]@                                |
++-----------------------------------+---------------------------------------------------------------------+
+| @'read' :: Read a => String -> a@ | @'Relude.String.Reexport.readMaybe' :: Read a => String -> Maybe a@ |
++-----------------------------------+---------------------------------------------------------------------+
+| @'fromJust' :: Maybe a -> a@      | @'Relude.Monad.Reexport.fromMaybe' :: a -> Maybe a -> a@            |
++-----------------------------------+---------------------------------------------------------------------+
+
+This module is intended to be imported qualified and it's not even
```
Comment (vrom911) [inline src/Relude/Unsafe.hs:29]:
```suggestion
This module is intended to be imported qualified and it is not
```

**Pull request merged**